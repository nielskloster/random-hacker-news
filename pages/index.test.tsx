import { describe, test, expect } from 'vitest'
import { render } from '@testing-library/react'
import IndexPage from '.'

describe('Index page', () => {
  test('App mounts properly', async () => {
    const wrapper = render(<IndexPage />)

    expect(wrapper).toMatchSnapshot()
  })
})