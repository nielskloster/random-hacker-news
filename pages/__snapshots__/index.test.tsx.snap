// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Index page > App mounts properly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header>
          <h1
            class="flex justify-center p-4 font-mono"
          >
            Random Hacker News
          </h1>
        </header>
        <div
          class="flex justify-center p-4 font-mono"
        >
          <div
            aria-label="loading"
            class="grid grid-cols-1 md:grid-cols-2 gap-4 p-y-12"
          >
            <div
              class="bg-slate-100 rounded-lg p-4 h-32 w-80"
            />
            <div
              class="bg-slate-100 rounded-lg p-4 h-32 w-80"
            />
            <div
              class="bg-slate-100 rounded-lg p-4 h-32 w-80"
            />
            <div
              class="bg-slate-100 rounded-lg p-4 h-32 w-80"
            />
            <div
              class="bg-slate-100 rounded-lg p-4 h-32 w-80"
            />
            <div
              class="bg-slate-100 rounded-lg p-4 h-32 w-80"
            />
            <div
              class="bg-slate-100 rounded-lg p-4 h-32 w-80"
            />
            <div
              class="bg-slate-100 rounded-lg p-4 h-32 w-80"
            />
            <div
              class="bg-slate-100 rounded-lg p-4 h-32 w-80"
            />
            <div
              class="bg-slate-100 rounded-lg p-4 h-32 w-80"
            />
          </div>
        </div>
        <footer />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header>
        <h1
          class="flex justify-center p-4 font-mono"
        >
          Random Hacker News
        </h1>
      </header>
      <div
        class="flex justify-center p-4 font-mono"
      >
        <div
          aria-label="loading"
          class="grid grid-cols-1 md:grid-cols-2 gap-4 p-y-12"
        >
          <div
            class="bg-slate-100 rounded-lg p-4 h-32 w-80"
          />
          <div
            class="bg-slate-100 rounded-lg p-4 h-32 w-80"
          />
          <div
            class="bg-slate-100 rounded-lg p-4 h-32 w-80"
          />
          <div
            class="bg-slate-100 rounded-lg p-4 h-32 w-80"
          />
          <div
            class="bg-slate-100 rounded-lg p-4 h-32 w-80"
          />
          <div
            class="bg-slate-100 rounded-lg p-4 h-32 w-80"
          />
          <div
            class="bg-slate-100 rounded-lg p-4 h-32 w-80"
          />
          <div
            class="bg-slate-100 rounded-lg p-4 h-32 w-80"
          />
          <div
            class="bg-slate-100 rounded-lg p-4 h-32 w-80"
          />
          <div
            class="bg-slate-100 rounded-lg p-4 h-32 w-80"
          />
        </div>
      </div>
      <footer />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
